cmd/docs/docs.go

```diff
@@ -0,0 +1,62 @@
+package docs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+)
+
+type Options struct {
+	// the path to generate the documentation in
+	path string
+}
+
+const DefaultPath = "./docs"
+
+func NewDocsCommand() *cobra.Command {
+	opts := &Options{}
+
+	return &cobra.Command{
+		Use:   "docs",
+		Short: "Generates the documentation for the CLI",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Use default path if no argument is provided
+			if len(args) == 0 {
+				opts.path = DefaultPath
+				fmt.Printf("No path was provided. Using default path: %s\n", DefaultPath)
+			} else {
+				absPath, err := filepath.Abs(args[0])
+				if err != nil {
+					return err
+				}
+				opts.path = absPath
+			}
+
+			// Create the directory if it doesn't exist
+			_, err := os.Stat(opts.path)
+			if os.IsNotExist(err) {
+				fmt.Printf("The directory %s does not exist. Creating it...\n", opts.path)
+
+				err := os.MkdirAll(opts.path, os.ModePerm)
+				if err != nil {
+					return fmt.Errorf("error creating documentation output directory %s: %s", opts.path, err)
+				}
+			}
+
+			// Generate markdown documentation
+			fmt.Printf("Generating documentation in %s...\n", opts.path)
+			err = doc.GenMarkdownTree(cmd, opts.path)
```

Comment:
Using this local cobra command will only generate docs for the `docs` command: you need to crawl back up the command tree to the root command to get docs generation for _all_ commands:  

```suggestion
			err = doc.GenMarkdownTree(cmd.Parent().Root(), opts.path)
```

Without this change:

```
❯ tree docs
docs
└── pizza_docs.md

1 directory, 1 file
```

With this change:

```
❯ tree docs
docs
├── pizza.md
├── pizza_completion.md
├── pizza_completion_bash.md
├── pizza_completion_fish.md
├── pizza_completion_powershell.md
├── pizza_completion_zsh.md
├── pizza_docs.md
├── pizza_generate.md
├── pizza_generate_codeowners.md
├── pizza_insights.md
├── pizza_insights_contributors.md
├── pizza_insights_repositories.md
├── pizza_insights_user-contributions.md
├── pizza_login.md
└── pizza_version.md

1 directory, 15 files
```

---

This suggested change would also break the tests you wrote since:

```go
		cmd := NewDocsCommand()
		err := cmd.Execute()
```

will panic since there's no parent or root for the command to crawl back up to.

In general, in my experience at least, using a `cobra.Command` directly in tests can get really really messy. If you have the full structure of the CLI (like with the root command) or things are well contained enough, it can work. But for this case, I'd say it's better to abstract out the things we want to test for (determining the file path) into its own function and test that.

Reply:
I was just coming here to mention that which is why I put it in draft mode. I figured I had to run it on the parent command. 😅

Reply:
For the testing, I was definitely not trying to test GenMarkdownTree, just trying to do a sanity check that markdown was generated, but given it runs at the root command now, I definitely agree it could get hairy since I'd probably have to mock a root command etc.

I've broken it up now to test folder creation, and like before, the assumption is GenMarkdownTree works (tested in Cobra I imagine).